kafka-rest/src/main/java/io/confluent/kafkarest/entities/ConsumerGroup.java

```diff
@@ -102,32 +105,51 @@ public abstract static class Builder {
   }
 
   public enum State {
-    UNKNOWN,
-
-    PREPARING_REBALANCE,
-
-    COMPLETING_REBALANCE,
-
-    STABLE,
-
-    DEAD,
+    UNKNOWN("Unknown"),
+    PREPARING_REBALANCE("PreparingRebalance"),
+    COMPLETING_REBALANCE("CompletingRebalance"),
+    STABLE("Stable"),
+    DEAD("Dead"),
+    EMPTY("Empty"),
+    ASSIGNING("Assigning"),
+    RECONCILING("Reconciling");
+
+    private static final Map<String, State> NAME_TO_ENUM =
+        Arrays.stream(values())
+            .collect(
+                Collectors.toMap(
+                    state -> state.name.toUpperCase(Locale.ROOT), Function.identity()));
+
+    private final String name;
+
+    State(String name) {
+      this.name = name;
+    }
 
-    EMPTY;
+    public static State parse(String name) {
+      State state = NAME_TO_ENUM.get(name.toUpperCase(Locale.ROOT));
+      return state == null ? UNKNOWN : state;
+    }
 
     public static State fromConsumerGroupState(GroupState state) {
       try {
-        return State.valueOf(state.name());
+        return State.parse(state.toString());
       } catch (IllegalArgumentException e) {
         return UNKNOWN;
       }
     }
 
-    public ConsumerGroupState toConsumerGroupState() {
+    public GroupState toConsumerGroupState() {
       try {
-        return ConsumerGroupState.valueOf(name());
+        return GroupState.parse(name);
       } catch (IllegalArgumentException e) {
-        return ConsumerGroupState.UNKNOWN;
+        return GroupState.UNKNOWN;
       }
     }
```

Comment:
How difficult is it to create a static map for this, instead of dynamically converting back and forth.